### ISSUE TITLE
The neighbor environments from CrystalNN can have additional stats

### ISSUE THREAD

**Issue by computron** [hackingmaterials/matminer#259]
Currently, the CrystalNN site fingerprint gives values that are essentially "volume invariant" (some of the chemical bonding cutoffs can change this, but the base algorithm behaves like this)

It would be nice to include some other statistics like "polyhedral volume" or "Voronoi statistics" or "average bond length" weighted by the different environments. e.g., if a site is 75% tetrahedral and 25% octahedral, to include the average bond length of the tetrahedral neighbors (weighted by 75%) and the average bond length of the octahedral neighbors (weighted by 25%) to produce an overall average bond length.

Basically, the idea is to bring in more information. This will be *not* useful in terms of classifying crystal prototypes, but should be useful in predicting gaps, bulk moduli, etc. that can scale with volume and thus require knowing volume information.
